polaris/ocean/tasks/geostrophic/analysis.py

```diff
@@ -0,0 +1,136 @@
+        """
+        Get the exact solution
+
+        Parameters
+        ----------
+        mesh_name : str
+            The mesh name which is the prefix for the initial condition file
+
+        field_name : str
+            The name of the variable of which we evaluate convergence
+            For the default method, we use the same convergence rate for all
+            fields
+
+        time : float
+            The time at which to evaluate the exact solution in seconds.
+            For the default method, we always use the initial state.
+
+        zidx : int, optional
+            The z-index for the vertical level at which to evaluate the exact
+            solution
+
+        Returns
+        -------
+        solution: xarray.DataArray
+            The exact solution with dimension nCells
+        """
+
+        if field_name not in ['h', 'normalVelocity']:
+            print(f'Variable {field_name} not available as an analytic '
+                  'solution for the geostrophic test case')
+
+        config = self.config
+
+        section = config['geostrophic']
+        alpha = section.getfloat('alpha')
+        vel_period = section.getfloat('vel_period')
+        gh_0 = section.getfloat('gh_0')
+
+        mesh_filename = f'{mesh_name}_mesh.nc'
+
+        h, _, _, normalVelocity = compute_exact_solution(
+            alpha, vel_period, gh_0, mesh_filename)
+
+        if field_name == 'h':
+            return h
+        else:
+            return normalVelocity
+
+    def get_output_field(self, mesh_name, field_name, time, zidx=None):
+        """
+        Get the model output field at the given time and z index
+
+        Parameters
+        ----------
+        mesh_name : str
+            The mesh name which is the prefix for the output file
+
+        field_name : str
+            The name of the variable of which we evaluate convergence
+
+        time : float
+            The time at which to evaluate the exact solution in seconds
+
+        zidx : int, optional
+            The z-index for the vertical level to take the field from
+
+        Returns
+        -------
+        field_mpas : xarray.DataArray
+            model output field
+        """
+
+        if field_name not in ['h', 'normalVelocity']:
+            print(f'Variable {field_name} not available for analysis in the '
+                  f'geostrophic test case')
+
+        if field_name == 'normalVelocity':
+            return super().get_output_field(mesh_name=mesh_name,
+                                            field_name=field_name,
+                                            time=time, zidx=zidx)
+        else:
+            ds_init = xr.open_dataset(f'{mesh_name}_init.nc')
+            bottom_depth = ds_init.bottomDepth
+            ssh = super().get_output_field(mesh_name=mesh_name,
+                                           field_name='ssh',
+                                           time=time, zidx=None)
+            h = ssh + bottom_depth
```

Comment:
In other cases with constant bottom_depth we just evaluate convergence with ssh. Are you just using wct for clarity or is there a reason I'm missing?

Reply:
I had a lot of trouble initially and only realize after some trial an error that it was because I was using a `z-level` rather than a `z-star` coordinate.  With `z-star`, you can have the bathymetry at zero, but I don't really love that approach.  It feels to me relatively important to use water-column thickness if that is the variable of interest and not to use `ssh` as a stand-in for water-column thickness.

Reply:
That's fair. Should we make gh_0 a function of `bottom_depth` then?

Reply:
I don't follow.  Isn't it related to water-column thickness, not elevation?

Reply:
Presumably when users change `bottom_depth` in the config file, `gh_0` should also be changed? But I could be misunderstanding the exact solution in the `_compute_h` function

Reply:
No, I don't think that's correct.  `_compute_h` should compute the water-column thickness, which should be independent of the choice of `bottom_depth`.  The `ssh` should be offset by the choice of `bottom_depth` such that `h` (water column thickness) remains the same.

Reply:
`h_0 = gh_0 / g` is part of the solution of h, but I don't think it is meant to be a reference elevation but rather a reference water-column thickness.

Reply:
But if my understanding isn't correct, it would be very important to know that now, since it could very much affect the error and convergence.

Reply:
I confirmed that changing the `bottom_depth` does not change the water column thickness, so we don't need to change `gh_0`.

Reply:
Okay, great, thank you for checking on that.  It makes me feel a lot more secure about it.